### PR TITLE
Create existing provider modal

### PR DESCRIPTION
**Pull request by jennchenn**
## Notion ticket link

[Create existing provider modal](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)

## Implementation description
* created a new common modal component for use for future modals
* the isOpen and onClose parameters are passed in from a useDisclosure hook in the parent calling the modal component so that the parent can register when the modal is called based on a button click
* currently an example is in the homepage for the existing provider modal

## Steps to test
1. go to branch
2. go to localhost:3000/ and scroll all the way down
3. at the bottom click the open modal button to see the modal for the existing providers

## What should reviewers focus on?
* if the existing provider modal should be imported somewhere else (i couldn't find the provider screen)
* if the modal looks as expected

Screen Shot 2022-11-11 at 5 51 35 PM

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/intake/ExistingProviderModal.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+              <Spacer />
+              <Button
+                size="sm"
+                variant="tertiary"
+                leftIcon={<Icon as={UserPlus} />}
+              >
+                Add new provider
+              </Button>
+            </Flex>
+            <Select placeholder="Select provider">
```
Comment (kabiroberai) [inline frontend/src/components/intake/ExistingProviderModal.tsx:43]: Just flagging this for your reference rn, but consider replacing this `Select` with a `CustomSelectField` once #88 is in.

Comment (jennchenn) [reply frontend/src/components/intake/ExistingProviderModal.tsx:43]: @kabiroberai does the `CustomSelectField` support multi-select?

File: frontend/src/components/common/ModalComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        <Text paddingBottom="62px" textStyle="header-medium">
+          {primaryTitle}
+        </Text>
+        <ModalCloseButton marginTop="40px" marginRight="20px" size="sm" />
+        {modalContent}
+        <ModalFooter paddingTop="68px">
+          <Button variant="tertiary" onClick={onClose}>
+            Cancel
+          </Button>
+          <Button variant="primary" disabled onClick={onClick}>
```
Comment (jennchenn) [inline frontend/src/components/common/ModalComponent.tsx:51]: once a user selects something, the button should no longer be disabled - i think we would need a way to determine whether the button should be disabled/not based on the contents

**Review by jennchenn — COMMENTED** [review]
looks good!!

two things:
1. we need to be able to control whether primary button is disabled/not
2. the select component should be a multi-select--this can be left for another ticket though

**Review by jennchenn — COMMENTED** [review]
there seems to be a bug with the disabled state: on first click the button is disabled on initial open; however if the user enters something, presses cancel, and reopens the modal, the button is still enabled--do you mind seeing if you can replicate this?

child provider

**Review by jennchenn — COMMENTED** [review]
disabled button bug is fixed - i'm unable to approve the PR bc i opened it but it looks good to me

**Review by juthikahoque — APPROVED** [review]

**Pull request merged**